warehouse/oidc/__init__.py

```diff
@@ -15,10 +15,22 @@
 
 
 def includeme(config):
+    oidc_provider_service_class = config.maybe_dotted(
+        config.registry.settings["oidc.backend"]
+    )
+
     config.register_service_factory(
         OIDCProviderServiceFactory(
-            provider="github", issuer_url="https://token.actions.githubusercontent.com"
+            provider="github",
+            issuer_url="https://token.actions.githubusercontent.com",
+            service_class=oidc_provider_service_class,
         ),
         IOIDCProviderService,
         name="github",
     )
+
+    # During deployments, we separate auth routes into their own subdomain
+    # to simplify caching exclusion.
+    auth = config.get_settings().get("auth.domain")
+
+    config.add_route("oidc.mint_token", "/api/oidc/mint-token", domain=auth)
```

Comment:
I'm not 100% sure this is where I want this route to live -- I don't think we have anything else at `/api`.

Reply:
Yeah, I agree -- I think we discussed putting it under the current integration route namespace; I can do that once this is in a slightly more stable state 🙂

Reply:
I'll think on this a bit, but also we should survey other providers to see if there's a common practice here.

Reply:
Just pinging back on this to say that I haven't forgotten; I plan to do this once we've done some review on the actual functionality 🙂 

(I think a route under `/integrations/` might be best?)

Reply:
@di any follow-up thoughts on the route path? It's still under `/api/` for now, but changing it is easy enough.

Reply:
I guess the question is whether this should be generic or GitHub-specific.

If it's GitHub-specific, which it looks like it is right now, this should probably be something like `/_/oidc/github/mint-token`

Reply:
Yeah, it's currently GitHub specific (since we need to select the right `OIDCProviderService` for the request). Ideally it'd be provider agnostic, but that would require partially inspecting the JWT before verifying it.